taglib/fileref.cpp

```diff
@@ -482,7 +499,7 @@ void FileRef::parse(IOStream *stream, bool readAudioProperties,
 {
   // Try user-defined resolvers.
 
-  d->file = detectByResolvers(stream->name(), readAudioProperties, audioPropertiesStyle);
+  d->file = detectByResolvers(stream, readAudioProperties, audioPropertiesStyle);
   if(d->file)
     return;
```

Comment:
If there is only a `FileTypeResolver` registered (and not a `StreamTypeResolver`!), `detectByResolvers()` will return 0 instead of detecting by file name. So I would keep the check by file type after checking by stream.

```
  // Try user-defined stream type resolvers.

  d->file = detectByResolvers(stream, readAudioProperties, audioPropertiesStyle);
  if(d->file)
    return;

  // Try user-defined file type resolvers.

  d->file = detectByResolvers(stream->name(), readAudioProperties, audioPropertiesStyle);
  if(d->file)
    return;
```

Reply:
Very good point, fixed